### ISSUE TITLE
This condition will always return 'false' since the types 'string | null' and 'number' have no overlap.deno-ts(2367)

### ISSUE THREAD

**Issue by BlackAsLight** [microsoft/TypeScript#50388]
# Bug Report

### 💻 Code

```ts
const nationID = 19818
if (localStorage.getItem("id") == nationID) {
    // do something
}
```

### 🙁 Actual behavior

Typescript incorrectly complains saying that there is no overlap here between a string | null and a number. If I was using three equal symbols then I would agree with it, but I’m not because I want it to do the conversion for me. "4" == 4 returns true while "4" === 4 returns false. It is wrong for typescript to treat a double equal symbol and triple equal symbol as the same. 

### 🙂 Expected behavior

Typescript to not not complain about something that isn’t true.

**Comment by MartinJohns** [comment]
Duplicate of #26592.
